torch/csrc/autograd/python_variable.h

```diff
@@ -24,14 +24,17 @@ THP_API PyObject *ParameterClass;
 bool THPVariable_initModule(PyObject *module);
 THP_API PyObject * THPVariable_Wrap(torch::autograd::Variable var);
 
-static inline bool THPVariable_CheckExact(PyObject *obj) {
-  auto obj_py_type = Py_TYPE(obj);
+static inline bool THPVariable_CheckExact(PyTypeObject* tp) {
   return (
-    obj_py_type == (PyTypeObject*)THPVariableClass ||
-    obj_py_type == (PyTypeObject*)ParameterClass
+    tp == (PyTypeObject*)THPVariableClass ||
+    tp == (PyTypeObject*)ParameterClass
   );
 }
 
+inline bool THPVariable_CheckExact(PyObject *obj) {
+  return THPVariable_CheckExact(Py_TYPE(obj));
+}
+
```

Comment:
I wouldn't do this. Sometimes, `PyTypeObject*` can be casted as a `PyObject*`, and call the wrong overload here (since they C-inherit from one another). In such a case, you may end up checking for the equivalent of `has_torch_function(type(type(obj)))`.

Reply:
Good catch!!! Serves me right for trying to be clever with overloads.

Reply:
Changed to be two different functions rather than an overload.